Sources/Moya/TargetType.swift

```diff
@@ -38,32 +32,39 @@ public extension TargetType {
     }
 }
 
-/// Represents a type of upload task.
-public enum UploadType {
+/// Represents an HTTP task.
+public enum Task {
 
-    /// Upload a file.
-    case file(URL)
+    /// A request with no additional data.
+    case requestPlain
 
-    /// Upload "multipart/form-data"
-    case multipart([MultipartFormData])
-}
+    /// A requests body set with data.
+    case requestData(Data)
 
-/// Represents a type of download task.
-public enum DownloadType {
+    /// A requests body set with encoded parameters.
+    case requestParameters(parameters: [String: Any], encoding: ParameterEncoding)
 
-    /// Download a file to a destination.
-    case request(DownloadDestination)
-}
+    /// A requests body set with data, combined with url parameters.
+    case requestCompositeData(bodyData: Data, urlParameters: [String: Any])
 
-/// Represents an HTTP task.
-public enum Task {
+    /// A requests body set with encoded parameters combined with url parameters.
+    case requestCompositeParameters(bodyParameters: [String: Any], bodyEncoding: ParameterEncoding, urlParameters: [String: Any])
+
+    /// A file upload task.
+    case uploadFile(URL)
+
+    /// A file upload task with extra parameters using the given encoding.
+    case uploadFileParameters(parameters: [String: Any], encoding: ParameterEncoding, URL)
```

Comment:
Also, not sure about that one. If you want to upload a file with parameters, you would just use `uploadMultipart`.